### PR TITLE
Adds aioboto3

### PR DESCRIPTION
**Pull request by dgasmith**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by skrsna** [comment]
thanks @dgasmith for adding this, happy to be a maintainer :)

File: recipes/aioboto3/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+requirements:
+  build:
+    - python >=3.7,<3.9
+    - pip
+    - poetry >=1.1.11,<2.0.0a
+    - poetry-dynamic-versioning >=0.13.1,<0.14.0a
+  run:
+    - python >=3.7
+    - aiobotocore ==1.4.2
```
Comment (skrsna) [inline recipes/aioboto3/meta.yaml:25]:
```suggestion
    - aiobotocore >=2.3.0
```
Needs ☝🏽 according to the build log

**Comment by dgasmith** [comment]
FYI @terrycain, we're going to build your fantastic package for the Conda-forge community.

File: recipes/aioboto3/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+about:
+  home: https://github.com/terrycain/aioboto3
+  summary: 'Wrapper to use boto3 resources with the aiobotocore async backend'
+  description: |
+    aiobotocore allows you to use near enough all of the boto3 client commands in an async manner just by prefixing the command with await.
+  home: https://github.com/terrycain/aioboto3
+  license: Apache-2.0
+  license_file: LICENSE
+  license_family: Apache
+  doc_url: https://aioboto3.readthedocs.io/en/latest/
```
Comment (markllama) [inline recipes/aioboto3/meta.yaml:49]: Add dev_url:
dev_url: https://github.com/terrycain/aioboto3

File: recipes/aioboto3/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    - python >=3.7,<3.9
+    - pip
+    - poetry >=1.1.11,<2.0.0a
+    - poetry-dynamic-versioning >=0.13.1,<0.14.0a
+  run:
+    - python >=3.7
+    - aiobotocore >=2.3.0
+    - cryptography >=2.3.1
+    - chalice >=1.24.0
+    - boto3 ==1.17.106
```
Comment (skrsna) [inline recipes/aioboto3/meta.yaml:28]:
```suggestion
    - boto3 >=1.24.21
```
Another upgrade according to build log

File: recipes/aioboto3/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    - s3transfer >=0.4.0,<0.5.0a
```
Comment (skrsna) [inline recipes/aioboto3/meta.yaml:29]:
```suggestion
    - s3transfer >=0.4.0,<0.7.0a
```
Just to be safe `v0.6.0` is the latest https://github.com/boto/s3transfer/releases/tag/0.6.0

File: recipes/aioboto3/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    - poetry >=1.1.11,<2.0.0a
+    - poetry-dynamic-versioning >=0.13.1,<0.14.0a
+  run:
+    - python >=3.7
+    - aiobotocore >=2.3.0
+    - cryptography >=2.3.1
+    - chalice >=1.24.0
+    - boto3 >=1.24.21
+    - s3transfer >=0.4.0,<0.7.0a
+    - idna >=2.10
```
Comment (markllama) [inline recipes/aioboto3/meta.yaml:30]:
```suggestion
  run:
    # base runtime environment
    - python >=3.7
    - pip

    # boto direct functional dependencies
    - aiobotocore >=2.3.0            # Async core layer for AWS python bindings
    - boto3 >=1.20.0                    # AWS API bindings for Python
    - s3transfer >=0.5.0,<0.6.0a  # python module to move files to and from AWS S3

    # General web service infrastructure modules
    - idna >=3.3                           # International Domain Names - for URL host formatting
    - cryptography >=37.0.0        # Python module that provide bindings for data encryption primatives
    - chalice >=1.27.0                 # serverless app framework - client side for comms to AWS API

```

File: recipes/aioboto3/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+about:
+  home: https://github.com/terrycain/aioboto3
+  summary: 'Wrapper to use boto3 resources with the aiobotocore async backend'
+  description: |
+    aiobotocore allows you to use near enough all of the boto3 client commands in an async manner just by prefixing the command with await.
+  home: https://github.com/terrycain/aioboto3
+  license: Apache-2.0
+  license_file: LICENSE
+  license_family: Apache
+  doc_url: https://aioboto3.readthedocs.io/en/latest/
```
Comment (markllama) [inline recipes/aioboto3/meta.yaml:49]: Add dev_url
```suggestion
  doc_url: https://aioboto3.readthedocs.io/en/latest/
  dev_url: https://github.com/terrycain/aioboto3
```

**Review by markllama — CHANGES_REQUESTED** [review]

File: recipes/aioboto3/meta.yaml (hunk 5 above)
Comment (dgasmith) [reply recipes/aioboto3/meta.yaml:30]: It's a little unusual to document these yaml so heavily.

**Comment by terrycain** [comment]
> FYI @terrycain, we're going to build your fantastic package for the Conda-forge community.

Awesome, any issues lemme know :)

File: recipes/aioboto3/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    - pip
+    - poetry >=1.1.11,<2.0.0a
+    - poetry-dynamic-versioning >=0.13.1,<0.14.0a
+  run:
+    # base runtime environment
+    - python >=3.7
+    - pip
+
+    # boto direct functional dependencies
+    - aiobotocore >=2.3.0            # Async core layer for AWS python bindings
```
Comment (markllama) [inline recipes/aioboto3/meta.yaml:29]: aioboto3 requires aibotocore pinned to 2.3.0
```suggestion
    - aiobotocore =2.3.0            # Async core layer for AWS python bindings
```

File: recipes/aioboto3/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+    # boto direct functional dependencies
+    - aiobotocore =2.3.0            # Async core layer for AWS python bindings
+    - boto3 >=1.20.0                    # AWS API bindings for Python
+    - s3transfer >=0.5.0,<0.6.0a  # python module to move files to and from AWS S3
+
+    # General web service infrastructure modules
+    - idna >=3.3                           # International Domain Names - for URL host formatting
+    - cryptography >=37.0.0        # Python module that provide bindings for data encryption primatives
+    - chalice >=1.27.0                 # serverless app framework - client side for comms to AWS API
```
Comment (skrsna) [inline recipes/aioboto3/meta.yaml:36]: chalice is only available for linux and osx https://anaconda.org/conda-forge/chalice

File: recipes/aioboto3/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: abac5dcfa871627b7040e6586a69e3359e2dfc4e15dc66135969f2d26fbdcb3b
+
+build:
+  number: 0
+  noarch: python
+  script: rm -f poetry.lock && {{ PYTHON }} -m pip install --no-deps .
```
Comment (skrsna) [inline recipes/aioboto3/meta.yaml:15]:
```suggestion
  script: rm -f poetry.lock && {{ PYTHON }} -m pip install --no-deps .
  skip: True  # [no win build for chalice]
```

File: recipes/aioboto3/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    - importlib-metadata # >=3.6
+    # boto direct functional dependencies
+    - aiobotocore =2.3.0            # Async core layer for AWS python bindings
+    - boto3 >=1.20.0                    # AWS API bindings for Python
+    - s3transfer >=0.5.0,<0.6.0a  # python module to move files to and from AWS S3
+
+    # General web service infrastructure modules
+    - idna # >=3.3                           # International Domain Names - for URL host formatting
+    - cryptography # >=37.0.0        # Python module that provide bindings for data encryption primatives
+    - chalice >=1.27.0                 # serverless app framework - client side for comms to AWS API
```
Comment (dgasmith) [inline recipes/aioboto3/meta.yaml:37]:
```suggestion
```

File: recipes/aioboto3/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+    - s3transfer >=0.5.0,<0.6.0a  # python module to move files to and from AWS S3
+
+    # General web service infrastructure modules
+    - idna # >=3.3                           # International Domain Names - for URL host formatting
+    - cryptography # >=37.0.0        # Python module that provide bindings for data encryption primatives
+
+
+test:
+  imports:
+    - aioboto3
```
Comment (dgasmith) [inline recipes/aioboto3/meta.yaml:41]:
```suggestion
    - aioboto3
    - aioboto3.s3
```

File: recipes/aioboto3/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+  requires:
+    - pip
+  commands:
+    - pip check
```
Comment (dgasmith) [inline recipes/aioboto3/meta.yaml:45]:
```suggestion
```

File: recipes/aioboto3/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+  number: 0
+  script: rm -f poetry.lock && {{ PYTHON }} -m pip install --no-deps .
+  skip: True  # [win]
+
+requirements:
+  build:
+    - python >=3.10
+    - pip
+    - poetry # >=1.1.11,<2.0.0a
+    - poetry-dynamic-versioning # >=0.13.1,<0.14.0a
```
Comment (dgasmith) [inline recipes/aioboto3/meta.yaml:22]:
```suggestion
    - poetry
    - poetry-dynamic-versioning
```

**Comment by markllama** [comment]
@conda-forge/staged-recipes, ready for review

**Comment by hmaarrfk** [comment]
Two questions:

1. Does this package contain any compiled code? If not, it should be "noarch".
2. If windows is skipped because a package doesn't exist for it. it is OK to mark the package as `noarch` and ignore the failure here. Users of windows won't be able to install the package due to the missing dependency. This ensures that you get your package for all unix combinations on day 1 without migrating to pypy, and then to osx arm64 and linux arm/ppc64le.

File: recipes/aioboto3/meta.yaml [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: abac5dcfa871627b7040e6586a69e3359e2dfc4e15dc66135969f2d26fbdcb3b
+
+build:
+  noarch: python
+  number: 0
+  script: rm -f poetry.lock && {{ PYTHON }} -m pip install --no-deps .
+  skip: True  # [win]
```
Comment (hmaarrfk) [inline recipes/aioboto3/meta.yaml:16]:
```suggestion
```

File: recipes/aioboto3/meta.yaml [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+
+requirements:
+
```
Comment (hmaarrfk) [inline recipes/aioboto3/meta.yaml:19]:
```suggestion
```

**Comment by markllama** [comment]
@hmaarrfk Thank you for the tips.

**Review by markllama — APPROVED** [review]

**Comment by markllama** [comment]
@conda-forge/staged-recipes ready for review again

**Comment by hmaarrfk** [comment]
Thanks

**Pull request merged**